drivers/sensor/qdec_s32k/qdec_s32k.c

```diff
@@ -0,0 +1,448 @@
+			    struct sensor_value *val)
+{
+	struct qdec_s32k_data *data = dev->data;
+
+	double rotation = (data->abs_counter * 2.0 * M_PI) / data->micro_ticks_per_rev;
+
+	switch (ch) {
+	case SENSOR_CHAN_ROTATION:
+		sensor_value_from_double(val, rotation);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api qdec_s32k_api = {
+	.sample_fetch = &qdec_s32k_fetch,
+	.channel_get = &qdec_s32k_ch_get,
+};
+
+static int qdec_s32k_initialize(const struct device *dev)
+{
+	const struct qdec_s32k_config *config = dev->config;
+	uint32_t rate;
+
+	if (config->trgmux_maps_len != 4 || config->trgmux_maps[0] > SIUL2_MAX_NUM_OF_IMCR_REG ||
+	    config->trgmux_maps[2] > SIUL2_MAX_NUM_OF_IMCR_REG) {
+		LOG_ERR("Wrong Trgmux config");
+		return -EINVAL;
+	}
+
+	pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+
+	if (Trgmux_Ip_Init(config->trgmux_config)) {
+		LOG_ERR("Could not initialize Trgmux");
+		return -EINVAL;
+	}
+
+	/* Assign eMIOS0 input to TRGMUX_OUT */
+	IP_SIUL2->IMCR[config->trgmux_maps[0]] = SIUL2_IMCR_SSS(config->trgmux_maps[1]);
+
+	/* Assign eMIOS0 input to TRGMUX_OUT */
+	IP_SIUL2->IMCR[config->trgmux_maps[2]] = SIUL2_IMCR_SSS(config->trgmux_maps[3]);
+
+	LOG_DBG("TRGMUX ACCESS Input[0] =%d Output[0]=%d",
+		config->trgmux_config->paxLogicTrigger[0]->Input,
+		config->trgmux_config->paxLogicTrigger[0]->Output);
+
+	if (Lcu_Ip_Init(config->lcu_config)) {
+		LOG_ERR("Could not initialize Lcu");
+		return -EINVAL;
+	}
+
+	/* Unmask relevant LCU OUT Channels */
+	Lcu_Ip_SyncOutputValueType EncLcuEnable[4U];
+
+	EncLcuEnable[0].LogicOutputId = LCU_LOGIC_OUTPUT_0;
+	EncLcuEnable[0].Value = 1U;
+	EncLcuEnable[1].LogicOutputId = LCU_LOGIC_OUTPUT_1;
+	EncLcuEnable[1].Value = 1U;
+	EncLcuEnable[2].LogicOutputId = LCU_LOGIC_OUTPUT_2;
+	EncLcuEnable[2].Value = 1U;
+	EncLcuEnable[3].LogicOutputId = LCU_LOGIC_OUTPUT_3;
+	EncLcuEnable[3].Value = 1U;
+	Lcu_Ip_SetSyncOutputEnable(EncLcuEnable, 4U);
+
+	Emios_Icu_Ip_SetInitialCounterValue(
+		config->emios_inst, config->emios_channels[EMIOS_CW_CH_IDX], (uint32_t)0x1U);
+	Emios_Icu_Ip_SetInitialCounterValue(
+		config->emios_inst, config->emios_channels[EMIOS_CCW_CH_IDX], (uint32_t)0x1U);
+
+	Emios_Icu_Ip_SetMaxCounterValue(config->emios_inst, config->emios_channels[EMIOS_CW_CH_IDX],
+					EMIOS_ICU_IP_COUNTER_MASK);
+	Emios_Icu_Ip_SetMaxCounterValue(config->emios_inst,
+					config->emios_channels[EMIOS_CCW_CH_IDX],
+					EMIOS_ICU_IP_COUNTER_MASK);
+
+	/* This API sets MCB/EMIOS_ICU_MODE_EDGE_COUNTER mode */
+	Emios_Icu_Ip_EnableEdgeCount(config->emios_inst, config->emios_channels[EMIOS_CW_CH_IDX]);
+	Emios_Icu_Ip_EnableEdgeCount(config->emios_inst, config->emios_channels[EMIOS_CCW_CH_IDX]);
+
+	clock_control_get_rate(config->clock_dev, config->clock_subsys, &rate);
+	return 0;
+}
+
+/* LCU Logic Input Configuration */
+#define LogicInputCfg_Common(MuxSel)								   \
+	{											   \
+		/* uint8 MuxSel		*/ MuxSel,						   \
```

Comment:
Instead of the comment `/* uint8 MuxSel */`, use the member name as `.MuxSel = MuxSel`. The S32 HAL is conditionally compiled and some members may not exist for some devices, so relying in the order is fragile. And also is self-documenting.

Reply:
fixed!